packages/backend/src/apps/formatter/actions/date-time/common/date-time-format.ts

```diff
@@ -0,0 +1,82 @@
+import type { IField } from '@plumber/types'
+
+import { DateTime } from 'luxon'
+import z from 'zod'
+
+/**
+ * This file contains stuff to handle all the date formats we want to support:
+ * - Field and schema definitions
+ * - Parsing functions
+ */
+
+interface DateFormatConverter {
+  parse: (input: string) => DateTime
+  stringify: (dateTime: DateTime) => string
+}
+
+const supportedFormats = z.enum(['formsgSubmissionTime', 'formsgDateField'])
+
+const formatConverters = {
+  formsgSubmissionTime: {
+    parse: (input: string): DateTime => DateTime.fromISO(input),
+    stringify: (dateTime: DateTime): string => dateTime.toISO(),
+  },
+  formsgDateField: {
+    parse: (input: string): DateTime =>
+      DateTime.fromFormat(input, 'dd MMM yyyy'),
+    stringify: (dateTime: DateTime): string => dateTime.toFormat('dd MMM yyyy'),
+  },
```

Comment:
bug(?): i had this issue where en-SG accepts `Sept` and not `Sep` so I had to manually check for en-US formats also i.e. 26 Sep 2023 will not work but 26 Sept 2023 will work currently but formSG passes `Sep` to us.

fix: what I did was to manually format both ways and check which passes in this code: https://github.com/opengovsg/plumber/blob/develop-v2/packages/backend/src/apps/delay/helpers/generate-timestamp.ts#L9-L28

Reply:
Thanks! I dug into the formsg source code and it looks like they always format to the en-US locale. Updated my code to handle.